### PR TITLE
Turn management

### LINKED ISSUE #21: player one turn
child of #20

### PR DESCRIPTION
**Pull request by anon0mys**
###### Changes Made:
* Adds user lookup by api-key
* Adds logic to ensure user making request is active player
* Changes game enum to 'player_1' and 'player_2'
* Updates ai to only fire if player 2 is nil

closes #21 #22 #23 #24 #25 #26

### CODE REVIEW

File: app/controllers/api/v1/games_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def create
       winner: nil,
       player_1_turns: 0,
       player_2_turns: 0,
-      current_turn: 'challenger'
+      current_turn: 'player_1'
```
Comment (Vadlusk) [inline app/controllers/api/v1/games_controller.rb]: I like the fact that they're all called player now makes me happy.

File: app/controllers/api/v1/games/shots_controller.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 class Api::V1::Games::ShotsController < ApiController
-  def create
-    game = Game.find(params[:game_id])
-    player = User.find_by(api_key: request.headers['X-API-Key'])
 
-    turn_processor = TurnProcessor.new(game, params[:shot][:target], player)
+  def create
+    @player = User.find_by(api_key: request.headers['X-API-Key'])
+    raise ApiExceptions::InvalidApiKey.new('Unauthorized', 401) if @player.nil?
```
Comment (jmejia) [inline app/controllers/api/v1/games/shots_controller.rb]: @anon0mys Think we can extract this `@player` concept to be a method within `ApiController`? If we reference a method instead it'll make testing easier and should also DRY up the code.

File: app/models/game.rb [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
-  enum current_turn: ["challenger", "computer"]
+  enum current_turn: ['player_1', 'player_2']
   serialize :player_1_board
   serialize :player_2_board
 
   validates :player_1_board, presence: true
   validates :player_2_board, presence: true
+
+  def active_player
+    if current_turn == 'player_1'
```
Comment (jmejia) [inline app/models/game.rb:12]: @anon0mys Could we just return `current_turn`?

Comment (jmejia) [reply app/models/game.rb:12]: Oh wait, nvm. Missed that it was a string. Try this...

`self.send(current_turn)`

File: app/controllers/api/v1/games/shots_controller.rb (hunk 2 above)
Comment (anon0mys) [reply app/controllers/api/v1/games/shots_controller.rb]: That is a great suggestion!

**Pull request merged**